### PR TITLE
🌲 Smart Salary Prediction using Random Forest Regression

### LINKED ISSUE #1634: Salary Prediction Using Random Forest
Hi @Niketkumardheeryan , I would like to contribute a **Salary Prediction using Random Forest Regression** project to ML-Capsule.

This project helps beginners understand:

* Data preprocessing
* Feature engineering
* Exploratory Data Analysis (EDA)
* Regression techniques
* Random Forest implementation
* Model evaluation and prediction

The project is built using real-world salary datasets and demonstrates how Machine Learning can be applied for salary estimation based on factors like experience, education, job role, and skills.

It includes:
Clean and beginner-friendly code
 Proper project structure
 Model training & evaluation
 Visualization and insights
Real-world ML workflow explanation

I believe this project would be valuable for learners exploring supervised learning and regression algorithms. Please assign this issue to me under GSSoC'26.

### PR DESCRIPTION
**Pull request by guruprasadvv**
Closes #1634

### Overview

Implemented a **Salary Prediction System using Random Forest Regression** that predicts employee salaries based on factors such as education, experience, job role, and location. The model leverages ensemble learning to improve prediction accuracy and handle complex, non-linear relationships within the dataset.

### Features Implemented

* 💼 Salary prediction using Random Forest Regressor
* 🧹 Data cleaning and preprocessing
* 🔄 Handling missing values and categorical feature encoding
* 📊 Exploratory Data Analysis (EDA) with visualizations
* 🌲 Machine Learning model training using Random Forest
* 📈 Model evaluation using MAE and RMSE metrics
* 🔮 Salary prediction for new user-provided inputs
* 🎨 User-friendly interface for input and prediction display

### Technical Implementation

* Performed data preprocessing and feature engineering
* Applied label encoding/feature transformation where required
* Trained and optimized a Random Forest Regression model
* Evaluated model performance using standard regression metrics
* Integrated prediction functionality into the application workflow
* Structured project files according to repository guidelines

### Benefits

* Provides reliable salary estimates based on multiple employee attributes
* Demonstrates practical application of machine learning in salary forecasting
* Handles non-linear relationships effectively through ensemble learning
* Reduces overfitting compared to traditional decision tree models
* Offers an intuitive and interactive user experience

### Tech Stack

* Python
* Pandas
* NumPy
* Matplotlib
* Scikit-learn
* Random Forest Regressor

### Testing

* Verified data preprocessing pipeline
* Tested model training and prediction accuracy
* Validated prediction outputs using sample datasets
* Evaluated model performance using MAE and RMSE
* Confirmed successful predictions for new input data

### CODE REVIEW

**Comment by guruprasadvv** [comment]
Hi @Niketkumardheeryan ,

I have completed the implementation for Issue #1634 and submitted the corresponding PR. The project includes salary prediction using a Random Forest Regressor, along with data preprocessing, EDA, model training, evaluation, and prediction functionality.
Thank you for your time and support.

**Comment by guruprasadvv** [comment]
Hi @Niketkumardheeryan ,

I have completed the implementation for Issue https://github.com/Niketkumardheeryan/ML-CaPsule/issues/1634 and submitted the corresponding PR. The project includes salary prediction using a Random Forest Regressor, along with data preprocessing, EDA, model training, evaluation, and prediction functionality.
Thank you for your time and support.

File: SALARY-PREDICTION-USING-RANDOM FOREST/expected_ctc.csv
Comment (Sibam-Paul) [inline SALARY-PREDICTION-USING-RANDOM FOREST/expected_ctc.csv:1]: unstage this file and add a link of this csv in the code base instead. use `gdown` in the codebase to fetch the csv

File: SALARY-PREDICTION-USING-RANDOM FOREST/codeoutput.ipynb
Comment (Sibam-Paul) [inline SALARY-PREDICTION-USING-RANDOM FOREST/codeoutput.ipynb:1]: link the csv url and run all cells .
add comments to make it look clean

File: SALARY-PREDICTION-USING-RANDOM FOREST/README.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# salary-prediction
+
```
Comment (Sibam-Paul) [inline SALARY-PREDICTION-USING-RANDOM FOREST/README.md]: follow this readme format :- Rainfall_Trend_Analysis_and_Forecasting/README.md

File: SALARY-PREDICTION-USING-RANDOM FOREST/codeforit
Comment (Sibam-Paul) [inline SALARY-PREDICTION-USING-RANDOM FOREST/codeforit:1]: remove this file only .**ipynb** file and **readme** accepted 
**Follow the pinned issues in the** [issue tracker](https://github.com/Niketkumardheeryan/ML-CaPsule/issues)

**Comment by Sibam-Paul** [comment]
@guruprasadvv make the changes proposed

**Review by Sibam-Paul — COMMENTED** [review]
@guruprasadvv please make the necessary changes

**Comment by guruprasadvv** [comment]
Hi @Sibam-Paul ,

Thank you for reviewing my PR. I have addressed all the requested review changes and pushed the updated commits. I noticed that I completed the requested revisions after a couple of days, and I wanted to ask if this PR is still eligible for any applicable GSSoC badges or recognition.

Regardless, I appreciate your time and feedback throughout the review process. Thank you!

File: salary-prediction-using-RF/salary_prediction_rf_fixed.ipynb
Comment (Sibam-Paul) [inline salary-prediction-using-RF/salary_prediction_rf_fixed.ipynb:1]: {3316A1B0-A482-4A84-BB8E-BEC0146A58E5}


looks like all cells are not run @guruprasadvv 

follow this PR format please #1952

**Review by Sibam-Paul — CHANGES_REQUESTED** [review]

**Comment by guruprasadvv** [comment]
Hi @Sibam-Paul,

I have addressed all the requested review changes and pushed the updated commits to this PR. The salary prediction notebook and README have been updated accordingly, and I ensured that unrelated project files were not modified in the final submission.

Could you please review the latest changes when you get a chance?

Additionally, I completed the requested revisions after a couple of days and wanted to check whether this PR would still be applicable for GSSoC badges or recognition.

Thank you for your time and feedback throughout the review process.

File: Flower_classification_android_app/gradle/wrapper/gradle-wrapper.properties
Comment (Sibam-Paul) [inline Flower_classification_android_app/gradle/wrapper/gradle-wrapper.properties:1]: What is this for ??

**Review by Sibam-Paul — CHANGES_REQUESTED** [review]
@guruprasadvv make changes 
Make it look professional

**Comment by guruprasadvv** [comment]
That is a file added by mistake into that folder sorry for the inconvenience occured . But I checked it and it's safe to merge.

**Review by Sibam-Paul — CHANGES_REQUESTED** [review]
Remove the file please

**Comment by guruprasadvv** [comment]
Sorry for the inconvenience occured @Sibam-Paul . This is my first time in contribution.Thank You for guiding me and it means a lot. I have made the changes.

**Comment by Sibam-Paul** [comment]
@guruprasadvv ur committing the deleted file ..raise the pr again correctly closing this
